File: CSharp/LINQtoDB/Models/Employee.cs
using LinqToDB;
using LinqToDB.Mapping;

namespace Models;

[Table("employees")]
public class Employee
{
    [PrimaryKey, Column("id"), DataType(dataType: DataType.Text)]
    public Guid Id { get; set; }

    [Column("position"), DataType(dataType: DataType.Text), NotNull]
    public Position Position { get; set; }

    [Column("department_id"), NotNull]
    public int DepartmentId { get; set; }
    public Department? Department { get; set; }

    [Column("person_id"), DataType(dataType: DataType.Text), NotNull]
    public Guid PersonId { get; set; }
    public Person? Person { get; set; }

    public Employee()
    {
        Id = Guid.NewGuid();
    }

    public override string ToString() => $"{{ Employee: {{ Id: {Id}, Person: {Person}, Department: {Department}, Position: {Position} }} }}";
}

public enum Position
{
    Staffer,
    Manager,
    Boss,
}

File: FSharp/LINQtoDB/Models/Employee.fs
namespace Models

open LinqToDB
open LinqToDB.Mapping
open System

type Position =
  | Boss = 0
  | Manager = 1
  | Staffer = 2

[<Table "employees">]
type Employee (position, department, person) =
  new () = Employee(Position.Staffer, Department (), Person ())

  [<PrimaryKey>][<Column "id">][<DataType (dataType=DataType.Text)>]
  member val Id = Guid.NewGuid () with get, set

  [<Column "position">][<DataType (dataType=DataType.Text)>][<NotNull>]
  member val Position = position with get, set

  [<Column "department_id">][<NotNull>]
  member val DepartmentId = 0 with get, set

  [<Association (ThisKey="department_id", OtherKey="id")>]
  member val Department = department with get, set

  [<Column "person_id">][<DataType (dataType=DataType.Text)>][<NotNull>]
  member val PersonId = Guid.Empty with get, set

  [<Association (ThisKey="person_id", OtherKey="id")>]
  member val Person = person with get, set

  with override this.ToString () =
        $"{{ Employee: {{ Id: {this.Id}, Person: {this.Person}, Department: {this.Department}, Position: {this.Position} }} }}"
